### DISCUSSION TITLE
Add dependency string to modal

### DISCUSSION THREAD

**Discussion by Sherlouk** [SwiftPackageIndex/SwiftPackageIndex-Server#1678] [Ideas]
So the site has the "Use this package" modal which makes it easy to copy the necessary string for the latest tag.

It would be awesome if the target dependency string was also copyable.

```
.product(name: "X", package: "Y"),
```

This may be a little difficult for packages where there are many products (like Firebase), but for packages where there's only one product (which I'd like to think is by far the majority) it would be of great help.

Currently this is a bit frustration as it still requires me to go to the GitHub repository, navigate to the Package.swift file, and grab the details from there.

**Comment by finestructure** [comment]
Yeah, I’ve missed that myself. I can never figure out the proper dependency and product statements without looking them up.

In principle, we’ve got all the data to populate even a list of `.product` clauses – we parse and store all the products we find in a manifest. The problem would be to make it available at the point where this is needed. The package page query doesn’t currently read from the products table.

**Comment by daveverwer** [comment]
We'd also need to think about the UI for this. [Some packages have a lot of products](https://swiftpackageindex.com/soto-project/soto).

**Reply by Sherlouk** [reply]
Absolutely, but these types of projects are rare.

I don't think it's completely unreasonable to put a sensible maximum on it, perhaps with a link to the Package.swift file if a users wishes to see the full list.

It's also potentially something which could utilise your JavaScript expansion mechanism to show more if they so choose.

Keep it simple to begin with though?

**Reply by daveverwer** [reply]
> Keep it simple to begin with though?

Of course. I'm afraid we still need to think about the UI in order to even implement a "simple" version, though!

What happens for packages that are not simple? Do they just not get the feature? Does it pick a product to show? If so, which one? If we set a maximum number of products, which products get included/excluded? Do we need the products next to each significant version, or is it good enough to just grab them from the default branch? If from the default branch, how are they presented alongside the three versions?

We should at least have a plan for how this would be presented inside that modal (which is already quite busy) before back-end work starts. It's not clear to me how we'd even present this for packages with *one* product right now.

I will start to have a think about it, but I'd welcome a discussion taking into account some of those questions.
